src/sources/mysql/mysql-schema.lisp

```diff
@@ -4,57 +4,29 @@
 
 (in-package :pgloader.source.mysql)
 
-;;;
-;;; Those functions are to be called from withing an already established
-;;; MySQL Connection.
-;;;
-;;; Handle MATERIALIZE VIEWS sections, where we need to create the views in
-;;; the MySQL database before being able to process them.
-;;;
-(defun create-my-views (views-alist)
-  "VIEWS-ALIST associates view names with their SQL definition, which might
-   be empty for already existing views. Create only the views for which we
-   have an SQL definition."
-  (unless (eq :all views-alist)
-    (let ((views (remove-if #'null views-alist :key #'cdr)))
-      (when views
-        (loop :for (name . def) :in views
-           :for sql := (destructuring-bind (schema . v-name) name
-                         (format nil
-                                 "CREATE VIEW ~@[~s.~]~a AS ~a"
-                                 schema v-name def))
-           :do
-           (log-message :info "MySQL: ~a" sql)
-           #+pgloader-image
-           (mysql-query sql)
-           #-pgloader-image
-           (restart-case
-               (mysql-query sql)
-             (use-existing-view ()
-               :report "Use the already existing view and continue"
-               nil)
-             (replace-view ()
-               :report "Replace the view with the one from pgloader's command"
-               (let* ((v-name   (cdr name))
-                      (drop-sql (format nil "DROP VIEW ~a;" v-name)))
-                 (log-message :info "MySQL: ~a" drop-sql)
-                 (mysql-query drop-sql)
-                 (mysql-query sql)))))))))
-
-(defun drop-my-views (views-alist)
-  "See `create-my-views' for VIEWS-ALIST description. This time we DROP the
-   views to clean out after our work."
-  (unless (eq :all views-alist)
-    (let ((views (remove-if #'null views-alist :key #'cdr)))
-      (when views
-        (let ((sql
-               (format nil "DROP VIEW ~{~a~^, ~};"
-                       (mapcar (lambda (qname)
-                                 (format nil "~@[~s.~]~a"
-                                         (car qname) (cdr qname)))
-                               (mapcar #'car views)))))
-          (log-message :info "MySQL: ~a" sql)
-          (mysql-query sql))))))
+(defclass copy-mysql (db-copy)
+  ((encoding :accessor encoding         ; allows forcing encoding
+             :initarg :encoding
+             :initform nil)
+   (range-list :accessor range-list
+               :initarg :range-list
+               :initform nil))
+  (:documentation "pgloader MySQL Data Source"))
+
+(defmethod initialize-instance :after ((source copy-mysql) &key)
+  "Add a default value for transforms in case it's not been provided."
+  (let ((transforms (and (slot-boundp source 'transforms)
+                         (slot-value  source 'transforms))))
```

Comment:
Why do we use `slot-boundp` and `slot-value` instead of a proper accessor? If I understand the code correctly, the `:initform` of that slot can be `NIL` if we want to have it empty, and this means that we will not have to ever check if it is bound - we just call the accessor, and if the result is non-`NIL`, we proceed.

Reply:
It's old code that not even needed anymore, I just killed it for good. Also it was me still discovering Common Lisp at the times I wrote that, so it's just patching it until it works rather than careful reasoning here.